src/commonMain/kotlin/com/ashampoo/kim/format/jpeg/JpegImageParser.kt

```diff
@@ -184,27 +147,27 @@ object JpegImageParser : ImageParser() {
         return JpegXmpParser.parseXmpJpegSegment(xmpSegments.first().segmentBytes)
     }
 
-    private fun getIptc(segments: List<Segment>): IptcMetadata? {
+    private fun getIptc(segments: List<JpegSegment>): IptcMetadata? {
 
         val app13Segments = segments.filterIsInstance<App13Segment>()
+        var iptcMetadata: IptcMetadata? = null
 
         for (segment in app13Segments) {
 
-            val iptcMetadata = try {
-                segment.parseIptcMetadata()
+            iptcMetadata = try {
+                val metadata = segment.parseIptcMetadata()
+                /*
+                 * In case of multiple APP13 segements the records should be merged.
+                 * It's now allowed to have multiple ones, if it does not fit into a single one.
+                 */
+                if (iptcMetadata != null && metadata != null)
+                    iptcMetadata + metadata
+                else
+                    metadata
             } catch (ignore: ImageReadException) {
                 println("Ignored broken IPTC.")
                 continue
             }
-
-            /* Take the first record. */
-            if (iptcMetadata != null)
-                return iptcMetadata
-
-            /*
-             * TODO In case of multiple APP13 segements the records should be merged.
-             *  It's now allowed to have multiple ones, if it does not fit into a single one.
-             */
         }
 
         return null
```

Comment:
The tests fail because after the removal of lines 201 & 202 this method always returns `null`. It should now return `iptcMetadata`.

You can run the tests with `./gradlew allTests`